src/olympia/addons/tests/test_views.py

```diff
@@ -709,8 +709,56 @@ def test_with_grouped_ratings(self):
         assert data == {'detail': 'show_grouped_ratings parameter should be a boolean'}
 
 
-class TestAddonViewSetCreate(UploadMixin, TestCase):
+class AddonViewSetCreateUpdateMixin:
+    SUCCESS_STATUS_CODE = 200
+
+    def request(self, **kwargs):
+        raise NotImplementedError
+
+    def test_set_contributions_url(self):
+        response = self.request(contributions_url='https://foo.baa/xxx')
+        assert response.status_code == 400, response.content
+        domains = ', '.join(amo.VALID_CONTRIBUTION_DOMAINS)
+        assert response.data == {
+            'contributions_url': [f'URL domain must be one of [{domains}].']
+        }
+
+        response = self.request(contributions_url='http://sub.flattr.com/xxx')
+        assert response.status_code == 400, response.content
+        assert response.data == {
+            'contributions_url': [
+                f'URL domain must be one of [{domains}].',
+                'URLs must start with https://.',
+            ]
+        }
+
+        valid_url = 'https://flattr.com/xxx'
+        response = self.request(contributions_url=valid_url)
+        assert response.status_code == self.SUCCESS_STATUS_CODE, response.content
+        assert response.data['contributions_url']['url'].startswith(valid_url)
```

Comment:
I think we can now use `==` instead of `startswith`.

Reply:
we do `startswith` because we append utm params to the end of the url